src/ahbicht/expressions/condition_nodes.py

```diff
@@ -91,8 +91,8 @@ class ConditionNode(ABC):
     )
 
 
-# TConditionNode is a type var that matches any class inheriting from ConditionNode (in contrast to Type[ConditionNode])
-TConditionNode = TypeVar("TConditionNode", bound=ConditionNode)
+# ConditionNodeType_co matches any class inheriting from ConditionNode (in contrast to Type[ConditionNode])
+ConditionNode_co = TypeVar("ConditionNode_co", bound=ConditionNode, covariant=True)
```

Comment:
What does the `covariant=True`

Reply:
https://peps.python.org/pep-0484/#covariance-and-contravariance